virttest/env_process.py

```diff
@@ -1035,16 +1051,13 @@ def preprocess(test, params, env):
             kvm_userspace_version = "Unknown"
     else:
         qemu_path = utils_misc.get_qemu_binary(params)
-        version_output = decode_to_text(a_process.system_output(
-            "%s -version" % qemu_path, verbose=False))
-        version_line = version_output.split('\n')[0]
-        matches = re.match(QEMU_VERSION_RE, version_line)
-        if matches:
-            kvm_userspace_version = "%s (%s)" % matches.groups()
-        else:
-            kvm_userspace_version = "Unknown"
+        kvm_userspace_version = _get_qemu_version(qemu_path)
+        qemu_dst_path = utils_misc.get_qemu_dst_binary(params)
+        if qemu_dst_path != qemu_path:
```

Comment:
My qemu_dst_path is None:

02:28:42 ERROR| Reproduced traceback from: /usr/local/lib/python3.6/site-packages/avocado_framework_plugins_vt-71.0-py3.6.egg/avocado_vt/test.py:438
02:28:42 ERROR| Traceback (most recent call last):
02:28:42 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugins_vt-71.0-py3.6.egg/virttest/error_context.py", line 135, in new_fn
02:28:42 ERROR|     return fn(*args, **kwargs)
02:28:42 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugins_vt-71.0-py3.6.egg/virttest/env_process.py", line 1058, in preprocess
02:28:42 ERROR|     _get_qemu_version(qemu_dst_path))
02:28:42 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugins_vt-71.0-py3.6.egg/virttest/env_process.py", line 88, in _get_qemu_version
02:28:42 ERROR|     "%s -version" % qemu_cmd, verbose=False))
02:28:42 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-71.0-py3.6.egg/avocado/utils/process.py", line 1168, in system_output
02:28:42 ERROR|     encoding=encoding)
02:28:42 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-71.0-py3.6.egg/avocado/utils/process.py", line 1035, in run
02:28:42 ERROR|     cmd_result = sp.run(timeout=timeout)
02:28:42 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-71.0-py3.6.egg/avocado/utils/process.py", line 908, in run
02:28:42 ERROR|     self._init_subprocess()
02:28:42 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-71.0-py3.6.egg/avocado/utils/process.py", line 622, in _init_subprocess
02:28:42 ERROR|     raise details
02:28:42 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-71.0-py3.6.egg/avocado/utils/process.py", line 619, in _init_subprocess
02:28:42 ERROR|     env=self.env)
02:28:42 ERROR|   File "/usr/lib64/python3.6/subprocess.py", line 729, in __init__
02:28:42 ERROR|     restore_signals, start_new_session)
02:28:42 ERROR|   File "/usr/lib64/python3.6/subprocess.py", line 1364, in _execute_child
02:28:42 ERROR|     raise child_exception_type(errno_num, err_msg, err_filename)
02:28:42 ERROR| FileNotFoundError: [Errno 2] No such file or directory: 'None' (None -version): 'None'

Here need `qemu_dst_path is not None`

Reply:
this issue is blocking all libvirt function testing.